### PR TITLE
Sync neo changes

### PR DESCRIPTION
**Pull request by superboyiii**
Related: https://github.com/neo-project/neo/pull/2044
Close https://github.com/neo-project/neo-node/issues/685
***Please dont't merge until related PRs are all merged.***

### CODE REVIEW

File: neo-cli/CLI/MainService.cs [hunk 1]
```diff
@@ -281,7 +281,7 @@ private byte[] LoadDeploymentScript(string nefFilePath, string manifestFilePath,
 
             // Build script
 
-            scriptHash = file.ScriptHash;
+            scriptHash = file.ToArray().ToHexString().HexToBytes().ToScriptHash();
```
Comment (shargon) [inline neo-cli/CLI/MainService.cs:284]:
```suggestion
            scriptHash = file.ToArray().ToScriptHash();
```

**Comment by shargon** [comment]
@superboyiii could you test it again? https://github.com/neo-project/neo/pull/1973

File: neo-cli/CLI/MainService.cs [hunk 2]
```diff
@@ -281,10 +281,11 @@ private byte[] LoadDeploymentScript(string nefFilePath, string manifestFilePath,
 
             // Build script
 
-            scriptHash = file.ScriptHash;
+            var fileData = file.ToArray();
+            scriptHash = fileData.ToScriptHash();
```
Comment (cloud8little) [inline neo-cli/CLI/MainService.cs:285]: this scripthash is incorrect.

Comment (Tommo-L) [reply neo-cli/CLI/MainService.cs:285]: see https://github.com/neo-project/neo-node/pull/680

**Comment by superboyiii** [comment]
> @superboyiii could you test it again?
Works well on my local within https://github.com/neo-project/neo/pull/2044 and https://github.com/neo-project/neo-devpack-dotnet/pull/391. Test with both native and nep17. @shargon

**Comment by superboyiii** [comment]
@shargon Need you review again. It works well on my local.

**Review by shargon — APPROVED** [review]

**Pull request merged**